pwndbg/gdblib/arch.py

```diff
@@ -96,7 +96,12 @@ def _get_arch(ptrsize: int):
     for match in ARCHS:
         if match in arch:
             # Distinguish between Cortex-M and other ARM
-            if match == "arm" and "-m" in arch:
+            # GDB will sometimes fail to correctly label Cortex-M binaries properly, and says it's simply 'arm'.
+            # Internally, GDB still detects the processes as Cortex-M, as it can access .xpsr, but it doesn't
+            # appear to expose this in information through any command/API. Since Cortex-M has the .xpsr flags register
+            # instead of .cpsr, we will check if it's present.
+            # See: https://github.com/pwndbg/pwndbg/issues/2153
+            if match == "arm" and ("-m" in arch or pwndbg.gdblib.regs.xpsr is not None):
```

Comment:
What is the value of `arch` if `-m` is in it? Shall we maybe add this to the comment just so that we don't loose the idea why this `-m in arch` check is here?

Reply:
Good cool - in one scenario, it can be `armv7e-m`. Adding a comment